stdlib/pathlib.pyi

```diff
@@ -222,6 +222,7 @@ class Path(PurePath):
     else:
         def write_text(self, data: str, encoding: str | None = None, errors: str | None = None) -> int: ...
     if sys.version_info < (3, 12):
+        @deprecated("Deprecated since Python 3.10 and scheduled for removal in Python 3.12. Use hardlink_to() instead.")
         def link_to(self, target: StrOrBytesPath) -> None: ...
```

Comment:
This warning is only actionable on Python 3.10+, since `hardlink_to()` was only added in Python 3.10. So I think we should maybe gate the deprecation warning here:

```suggestion
    if sys.version_info < (3, 12):
        if sys.version_info >= (3, 10):
	        @deprecated("Deprecated since Python 3.10 and scheduled for removal in Python 3.12. Use hardlink_to() instead.")
	        def link_to(self, target: StrOrBytesPath) -> None: ...
	    else:
	        def link_to(self, target: StrOrBytesPath) -> None: ...
```